src/features/VisualNovel/pages/MediaPage/MediaPage.tsx

```diff
@@ -18,21 +17,18 @@ export const MediaPage: VFC = () => {
 
   const settingsContext = useSettingsContext();
 
-  /** Filter screenshots by nsfw flag. */
-  const filterPredicate = (screen: VisualNovelScreenshot): boolean => (settingsContext.isNsfwContentAllowed ? true : !screen.isNsfw);
-
   const filteredScreens = useMemo(() => {
     if (data) {
-      return data.screens.filter(filterPredicate);
+      return data.screens.filter(screen => (settingsContext.isNsfwContentAllowed ? true : !screen.isNsfw));
     }
     return [];
-  }, [data, filterPredicate]);
+  }, [data, settingsContext.isNsfwContentAllowed]);
 
   /**
-   * Handles clicking image.
+   * Handles click on an image.
    * @param index Image clicked index.
    */
-  const handleClickOnImage = useCallback((index: number) => {
+  const handleImageClick = useCallback((index: number) => {
```

Comment:
```suggestion
  const handleImageClick = useCallback((index: number) => () => {
```
To properly pass it to `onClick`